Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptions.swift

```diff
@@ -4,120 +4,114 @@
 
 import Foundation
 
+public typealias ChatMessageLayoutOptions = Set<ChatMessageLayoutOption>
+
+public extension ChatMessageLayoutOptions {
+    var rawValue: String {
+        map(\.rawValue).joined(separator: "-")
+    }
+
+    mutating func remove(_ options: ChatMessageLayoutOptions) {
+        self = subtracting(options)
+    }
+
+    mutating func insert(_ options: ChatMessageLayoutOptions) {
+        options.forEach { self.insert($0) }
+    }
+
+    /// Returns all options the current option set consists of separated by `-` character.
+    var description: String {
+        Self.singleOptions
+            .compactMap { contains($0) ? $0.optionName : nil }
+            .joined(separator: "-")
+    }
+
+    static let singleOptions: [ChatMessageLayoutOption] = [
+        .flipped,
+        .bubble,
+        .continuousBubble,
+        .avatarSizePadding,
+        .avatar,
+        .timestamp,
+        .authorName,
+        .text,
+        .quotedMessage,
+        .threadInfo,
+        .errorIndicator,
+        .reactions,
+        .onlyVisibleForYouIndicator,
+        .centered
+    ]
+}
+
 /// Describes the layout of base message content view.
-public struct ChatMessageLayoutOptions: OptionSet, Hashable {
-    public let rawValue: Int
+public struct ChatMessageLayoutOption: RawRepresentable, Hashable, Equatable, ExpressibleByStringLiteral {
+    public typealias StringLiteralType = String
 
-    public init(rawValue: Int) {
+    public let rawValue: String
+
+    public init(rawValue: String) {
         self.rawValue = rawValue
     }
+
+    public init(stringLiteral value: String) {
+        rawValue = value
+    }
+
+    @available(*, deprecated, message: "Use the string raw value initialiser.")
+    public init(rawValue: Int) {
+        self.rawValue = "\(rawValue)"
+    }
+
+    var optionName: String? {
```

Comment:
Why is this optional? rawValue is not optional

Reply:
The previous API had this optional. It was just to not add a breaking change. But I noticed now that his was not exposed before, so I will actually remove it since we don't need it 👍

Reply:
Removed the `optionName` ✅